Packs/PrismaCloud/Integrations/RedLock/RedLock.py

```diff
@@ -881,18 +900,26 @@ def fetch_incidents():
     demisto.info("Executing Prisma Cloud (RedLock) fetch_incidents with payload: {}".format(payload))
     response = req('POST', 'alert', payload, {'detailed': 'true'})
     incidents = []
+
+    fetched_ids[now] = []
+
     for alert in response:
-        if any(alert.get('id') in x for x in fetched_ids):
-            demisto.debug(f"Fetched {alert.get('id')} already. Skipping")
+        alert_id = alert.get('id')
+        if any(alert_id in existing_fetched_ids for existing_fetched_ids in fetched_ids.values()):
+            demisto.debug(f"Fetched {alert_id} already. Skipping")
             continue
-        demisto.debug(f"{alert.get('id')} has not been fetched. Processing.")
+
+        demisto.debug(f"{alert_id} has not been fetched. Processing.")
         incidents.append({
-            'name': alert.get('policy.name', 'No policy') + ' - ' + alert.get('id'),
+            'name': alert.get('policy.name', 'No policy') + ' - ' + alert_id,
             'occurred': convert_unix_to_demisto(alert.get('alertTime')),
             'severity': translate_severity(alert),
             'rawJSON': json.dumps(alert)
         })
-        fetched_ids.append({alert.get('id'): now})
+        fetched_ids[now].append(alert_id)
+
+    if not fetched_ids[now]:  # if no new incidents were added, no need to keep the date, saving space
+        fetched_ids.pop(now, None)
```

Comment:
Maybe we can do it the opposite way- save to a list of `fetched_ids_now` and if it is not empty, save this list to `fetched_ids[now]`. But both are good.